src/Limiters/limiter.jl

```diff
@@ -0,0 +1,119 @@
+"""
+    quasimonotone_limiter!(ρq, ρ, min_ρq, max_ρq; rtol)
+
+Arguments:
+- `ρq`: tracer density Field, where `q` denotes tracer concentration (or mixing
+  ratio) per unit mass
+- `ρ: fluid density Field
+- `min_ρq: Array of min(ρq) per element
+- `max_ρq`: Array of max(ρq) per element
+- `rtol`: relative tolerance needed to solve element-wise optimization problem
+
+This limiter is inspired by the one presented in Guba et al [GubaOpt2014](@cite).
+In the reference paper, it is denoted by OP1, and is outlined in eq.s (37)-(40).
+Quasimonotone here is meant to be monotone with respect to the spectral element
+nodal values. This limiter involves solving a constrained optimization problem
+(a weighted least square problem up to a fixed tolerance denoted by `rtol`)
+that is completely local to each element.
+As in HOMME, the implementation idea here is the following: we need to find a grid
+field which is closest to the initial field (in terms of weighted sum), but satisfies
+the min/max constraints. So, first we find values which do not satisfy constraints
+and bring these values to a closest constraint. This way we introduce some mass
+change (`mass_change`), so, we redistribute `mass_change` in the way that the
+l2 error is smallest.This redistribution might violate constraints thus, we do
+a few iterations.
```

Comment:
How many?

Reply:
In the unit test, where I start with a value that is 10% off the required bound, it converges in two iterations

Reply:
Actually, I start 20% off in the unit test (value -0.2 with bounds [0,1])

Reply:
Maybe add a comment that with an rtol of xx (value you use), it typically converges in a couple iterations?